src/loaders/tiff/ome-tiff.ts

```diff
@@ -0,0 +1,71 @@
+import type { GeoTIFF } from 'geotiff';
+import { fromString } from '../omexml';
+
+import TiffPixelSource from './pixel-source';
+import {
+  getLegacyIndexer,
+  getSubIFDIndexer,
+  OmeTiffIndexer
+} from './lib/indexers';
+import { getPixelSourceMeta } from './lib/utils';
+
+export interface OmeTiffSelection {
+  t: number;
+  c: number;
+  z: number;
+}
+
+export async function load(tiff: GeoTIFF) {
+  // Get first image from tiff and inspect OME-XML metadata
+  const firstImage = await tiff.getImage(0);
+  const {
+    ImageDescription,
+    SubIFDs,
+    PhotometricInterpretation: photometricInterpretation
+  } = firstImage.fileDirectory;
+  const omexml = fromString(ImageDescription);
+
+  /*
+   * Image pyramids are stored differently between versions of Bioformats.
+   * Thus we need a different indexer depending on which format we have.
+   */
+  let levels;
+  let pyramidIndexer: OmeTiffIndexer;
+
+  if (SubIFDs) {
+    // Image is >= Bioformats 6.0 and resolutions are stored using SubIFDs.
+    levels = SubIFDs.length;
+    pyramidIndexer = getSubIFDIndexer(tiff, omexml);
+  } else {
+    // Image is legacy format; resolutions are stored as separate images.
+    levels = omexml.length;
+    pyramidIndexer = getLegacyIndexer(tiff, omexml);
+  }
+
+  // TODO: The OmeTIFF loader only works for the _first_ image in the metadata.
+  const imgMeta = omexml[0];
```

Comment:
We should chat about this.  I would like to resolve this in this PR since it seems our options are somewhat limited - I only see making the image selection part of the `loadOmeTiff` method (one loader to one OME image) or making it part of the loader selection (i.e one loader can represent many OME images).  I don't think it makes a huge difference but I would lean to the former for making it configurable.  Definitely would like to hear your take.  The flower ome-tiff that you referenced in that comment about changing the z-stack has multiple imaging angles saved as multiple OME images so I think it could serve as good test data.

Reply:
I think this would be more appropriate for a different PR. `loadBioformatsZarr` has the same exact behavior, and loading other images within the TIFFs would require changes to the indexer code. 

> or making it part of the loader selection (i.e one loader can represent many OME images)

I think this approach would negate the effort put forward in this PR. The point of the PR is to provide an abstraction (`PixelSource`) that reflects an _single_ stack of images at a resolution. Other images in the tiff would equally be represented as separate PixelSources. My thought is that we can extend the `options` for `loadOmeTiff` to with a `loadAllPixelSources` flag or something. 

In that case, the object return type wouldn't be `{ data, metadata }` but `{ data, metadata }[]`.

Reply:
I'm ok with that.  Going to tag this in the issue for future reference, I like the plan but I think that it should be possible still to load only one of the images (instead of having to load all).  Maybe I misunderstand and you are ok with that in the future.  In any case, I'm ok with this.

Reply:
Right, I think we can decide on an api later. For now, just get the first image as this was the previous behavior. Later, we can add support to load all or some of the other images (by index for example). Something like:

```javascript
loadOmeTiff(url, { images: [0, 2, 3] })
```

Where `images` is optional and if provided we return an array of `{ data, metadata}` objects. Otherwise we just default to first image

Reply:
We can spend time thinking about that API, but I think that conversation belongs outside of this PR.